### DISCUSSION TITLE
CSS and JS file per page

### DISCUSSION THREAD

**Discussion by chiqui3d** [vercel/next.js#11133] [Help]
When I generate my static html pages, I see that both on the home page and in the privacy policy, it is adding the same js and css files, is there any way to control that?

I apologize if this is a very basic question.

**Comment by timneutkens** [comment]
Not exactly sure what you're referring to, can you add more details?

**Reply by chiqui3d** [reply]
Thank you very much for responding. I just tried importing a `JS `file into a page and it works, it is only included in that page. Perfect!!

But with `CSS`, it doesn't. The only way I found is using `CSS modules`, but for me it's not a good idea, since it forces you to use and import the `classes` as if the  was an `object`, and that's not the way I work with CSS.  Ideally, if you include a file such as `privacy.module.scss`, you don't have to use it as if it were a Javascript object.  I say this because for people who come from other ways of working it will be easier to migrate from one system to another if it act the way i said before. A clear example of your philosophy, and that's why I chose `Nextjs `is because it doesn't force me to work with GraphQL, the same should happen with CSS.

In short, the ideal would be that by including a CSS file in a component or a page, it only include it where it is needed, `just as CSS Modules works`, but without having to use CSS class as if it were an `object `and thus abstract it and with fewer dependencies.


Thank you, great work and excuse my English.

**Reply by chiqui3d** [reply]
Actually, I don't know if what I'm requesting can be done now.

**Reply by timneutkens** [reply]
You can't import global CSS in pages for the simple reason that it will cause many ordering issues when the CSS is not properly prefixed.

Basically if you move between pages using `next/link` you will get wildly different results than when just accessing the page.

You can add a global stylesheet that is not prefixed automatically in this way: https://nextjs.org/docs/basic-features/built-in-css-support#adding-a-global-stylesheet

**Reply by chiqui3d** [reply]
Thanks. I think I can get used to CSS modules, or if not make some `HOC`, such as `withCSSPerPage`, and find in the css folder a file that is called the same as the page and add it to the head. Example: `privacy.js and privacy.css|scss`

**Reply by chiqui3d** [reply]
Anyway, I think there is no need to prefix those CSS, just generate a specific CSS file for that page and place it in the last place of the CSS links.

**Reply by timneutkens** [reply]
The reason you need css modules is that styles have a 99% chance of conflicting if they're global when you move between pages client-side.
